lib/ansible/plugins/lookup/password.py

```diff
@@ -110,7 +110,7 @@
 
 - name: create random but idempotent password
   set_fact:
-    password: "{{ lookup('password', '/dev/null', seed=inventory_hostname) }}"
+    password: "{{ lookup('password', '/dev/null seed=' + inventory_hostname) }}"
```

Comment:
I wonder if this should actually be fixed or at least be an error. In the original example, seed is passed in as a kwarg, but is never used during option setting `self.set_options(var_options=variables, direct=kwargs)`.

Reply:
this was one of the lookups i missed during 'conversion', it should be "simpleish" to fix to use get_option when setting params[] dict.

Reply:
```diff --git a/lib/ansible/plugins/lookup/password.py b/lib/ansible/plugins/lookup/password.py
index 3eb48280ea..fb98d52a3f 100644
--- a/lib/ansible/plugins/lookup/password.py
+++ b/lib/ansible/plugins/lookup/password.py
@@ -175,7 +175,7 @@ def _parse_parameters(term):
     params['length'] = int(params.get('length', DEFAULT_LENGTH))
     params['encrypt'] = params.get('encrypt', None)
     params['ident'] = params.get('ident', None)
-    params['seed'] = params.get('seed', None)
+    params['seed'] = params.get('seed', self.get_option('seed'))

     params['chars'] = params.get('chars', None)
     if params['chars']:
@@ -337,6 +337,8 @@ class LookupModule(LookupBase):
     def run(self, terms, variables, **kwargs):
         ret = []

+        self.set_options(var_options=variables, direct=kwargs)
+
         for term in terms:
             relpath, params = _parse_parameters(term)
             path = self._loader.path_dwim(relpath)```

Reply:
and the same for the rest of the params